File: 강재연/programmers/P181835.js
// 문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/181835
// 시간: 0.04 ms
// 메모리: 33.5 MB

function solution(arr, k) {
    return arr.map( num => (k%2 == 0 ? num+k : num *k));
}

File: 공예진/programmers/P181835.java
//문제: https://school.programmers.co.kr/learn/courses/30/lessons/181835
//시간: 5.69ms
//메모리: 180MB

class Solution {
    public int[] solution(int[] arr, int k) {
        if (k % 2 != 0) {
            for (int i = 0; i  < arr.length; i++) {
                arr[i] *= k;
            }
        } else {
            for (int i = 0; i  < arr.length; i++) {
                arr[i] += k;
            }
        }
        return arr;
    }
}